src/companion_ad.js

```diff
@@ -1,22 +1,27 @@
-export class CompanionAd {
-  constructor(creativeAttributes = {}) {
-    this.id = creativeAttributes.id || null;
-    this.width = creativeAttributes.width || 0;
-    this.height = creativeAttributes.height || 0;
-    this.assetWidth = creativeAttributes.assetWidth || null;
-    this.assetHeight = creativeAttributes.assetHeight || null;
-    this.expandedWidth = creativeAttributes.expandedWidth || null;
-    this.expandedHeight = creativeAttributes.expandedHeight || null;
-    this.apiFramework = creativeAttributes.apiFramework || null;
-    this.adSlotID = creativeAttributes.adSlotID || null;
-    this.staticResources = [];
-    this.htmlResources = [];
-    this.iframeResources = [];
-    this.adParameters = null;
-    this.xmlEncoded = null;
-    this.altText = null;
-    this.companionClickThroughURLTemplate = null;
-    this.companionClickTrackingURLTemplates = [];
-    this.trackingEvents = {};
-  }
+export function createCompanionAd(creativeAttributes = {}) {
+  return {
+    id: creativeAttributes.id || null,
+    adType: 'companionAd',
+    width: creativeAttributes.width || 0,
+    height: creativeAttributes.height || 0,
+    assetWidth: creativeAttributes.assetWidth || null,
+    assetHeight: creativeAttributes.assetHeight || null,
+    expandedWidth: creativeAttributes.expandedWidth || null,
+    expandedHeight: creativeAttributes.expandedHeight || null,
+    apiFramework: creativeAttributes.apiFramework || null,
+    adSlotID: creativeAttributes.adSlotID || null,
+    staticResources: [],
+    htmlResources: [],
+    iframeResources: [],
+    adParameters: null,
+    xmlEncoded: null,
+    altText: null,
+    companionClickThroughURLTemplate: null,
+    companionClickTrackingURLTemplates: [],
+    trackingEvents: {}
+  };
+}
+
+export function isCompanionAd(companionAd) {
```

Comment:
you named the param `companionAd` but it might not be one. I would rename it to `ad` or `adVariation`

Reply:
same for the other functions